channeldb/invoice_test.go

```diff
@@ -351,28 +373,27 @@ func TestInvoiceAddTimeSeries(t *testing.T) {
 		}
 	}
 
+	var settledInvoices []Invoice
+	var settleIndex uint64 = 1
 	// We'll now only settle the latter half of each of those invoices.
 	for i := 10; i < len(invoices); i++ {
 		invoice := &invoices[i]
 
 		paymentHash := invoice.Terms.PaymentPreimage.Hash()
 
 		_, err := db.UpdateInvoice(
-			paymentHash, getUpdateInvoice(0),
+			paymentHash, getUpdateInvoice(invoice.Terms.Value),
 		)
 		if err != nil {
 			t.Fatalf("unable to settle invoice: %v", err)
 		}
-	}
 
-	invoices, err = db.FetchAllInvoices(false)
-	if err != nil {
-		t.Fatalf("unable to fetch invoices: %v", err)
-	}
+		// Create the settled invoice for the expectation set.
+		settleTestInvoice(invoice, settleIndex)
+		settleIndex++
 
-	// We'll slice off the first 10 invoices, as we only settled the last
-	// 10.
-	invoices = invoices[10:]
+		settledInvoices = append(settledInvoices, *copyInvoice(invoice))
```

Comment:
Is this copy necessary?

Reply:
Same question about other `copyInvoice` calls in test.

Reply:
Removed as these are indeed not necessary. (Imo it's a bit cleaner to copy this way as we guarantee immutability if the test is refactored later though.)